_posts/2015-03-stitch-fix-heart-unix.md

```diff
@@ -0,0 +1,139 @@
+*Proposed Outline:*
+```
+I. Introduction
+II. Table of contents by problem
+III. Each problem
+  A. Title
+    1. Attribution and environment sub-heading
+  B. Problem
+  C. Solution
+    1. Input
+    2. Command
+    3. Explanation
+IV. ~~Closing~~?
+```
+
+--
+
+I &#10084; UNIX and using the command line; they help me solve problems at Stitch Fix.  I'm not alone.  Across the Data Science and Engineering teams, we're constantly solving problems with UNIX and the command line.  **TODO: Closing sentence replaces closing section?**  
+
+   
+## Problems and Solutions
+ - Files
+   - [What size are these images?](#check-image-size)
+   - [What files recently changed?]()
+   - [How many lines are in this data dump?]()
+   - [How do you bulk change filenames?]()
+   - [Can I efficiently and securely copy these files?]()
+ - Directories
+   - [Directory File Summary]()
+   - [Sort Directories By Number Of Files]()
+   - [Delete Directory With Large Number Of Files]()
+ - CSVs
+   - [Filter CSV File By Column Values]()
+   - [Index CSV File]()
+ - Systems 
+   - [Display Resource Usage and Availability]()
+   - [Easily Manage Python Virtual Environments]()
+ - Productivity
+   - [Alias Everything]()
+   - [Command Completion Alert]()
+ - [Bemuse Coworkers]()
+
+<a name="check-image-size"></a>
+### Check Image Size
+&#8212; [Dave Copeland](https://twitter.com/davetron5000/) *(OS X / bash)*
+
+#### Problem
+Check the sizes of our product images the pickers use to check what they are picking in our warehouse
+
+#### Solution
+Created a script to run curl on the argument to download the image, use ImageMagick to get its size and print that out in CSV.  Piped my input CSV of images into `xargs -n1 -P8 ./my_script.rb` to basically run my script 8-way parallel to get the job done as fast as I could without setting my machine on fire.
+
```

Comment:
Yeah, these should probably be cleaned up.  I wrote mine in haste.

Depending on the amount of work you want to put in, you could use our input as just input and write it in your own voice, e.g. "Dave had to do X, and so used xargs to do it" or whatever.  That could be a lot of work tho :)